### PR TITLE
Developer guide update with virtual env and additional services

### LINKED ISSUE #2257: Development Guide updates for working with local user and all dependency services
Currently, the development guide suggested working with globally installed `intelmq`. For lowering potential conflicts with the local machine, easier development and integrations with IDE, I'd propose a guide to develop using a virtual environment and local non-root user.

In addition, there are some missing information about commonly used local services and test packages. I'd like to propose guide updates and eventually some additional automaton (e.g. docker-compose with development services) to make the start easier and quicker.

### PR DESCRIPTION
**Pull request by kamil-certat**
Improvements to the development guide to make the dev configuration easier.

Closes: #2257

### CODE REVIEW

**Comment by kamil-certat** [comment]
@sebix @waldbauer-certat May I ask for rerun tests? They should pass now

**Comment by sebix** [comment]
> @sebix @waldbauer-certat May I ask for rerun tests? They should pass now

I re-ran the job

**Comment by kamil-certat** [comment]
Thanks, when you time had, it would be nice to see your opinion on changes :)

File: docs/dev/guide.rst [hunk 1]
```diff
@@ -72,7 +83,43 @@ In this guide we use `/opt/dev_intelmq` as local repository copy. You can also u
...
+
+   python -m venv .venv
+   source .venv/bin/activate
+
+   pip install -e .
+
+   # It creates an empty folder and assign group and owner for the current user
+   sudo install -g `whoami` -o `whoami` -d /opt/intelmq
+
+   intelmqsetup --skip-ownership
```
Comment (sebix) [inline docs/dev/guide.rst]: Could the owner and group instead be set by `intelmqsetup`?

File: docs/dev/guide.rst [hunk 2]
```diff
@@ -72,7 +83,43 @@ In this guide we use `/opt/dev_intelmq` as local repository copy. You can also u
...
    intelmqsetup
 
-**Note:** please do not forget that configuration files, log files will be available on `/opt/intelmq`. However, if your development is somehow related to any shipped configuration file, you need to apply the changes in your repository `/opt/dev_intelmq/intelmq/etc/`.
+
+Using virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+   git clone https://github.com/<your username>/intelmq.git /opt/dev_intelmq
```
Comment (sebix) [inline docs/dev/guide.rst]: This assumes, that `/opt` is writeable by the current user

File: docs/dev/guide.rst [hunk 3]
```diff
@@ -57,7 +57,18 @@ Developers can create a fork repository of IntelMQ in order to commit the new co
 The following instructions will use `pip3 -e`, which gives you a so called *editable* installation. No code is copied in the libraries directories, there's just a link to your code. However, configuration files still required to be moved to `/opt/intelmq` as the instructions show.
 
 In this guide we use `/opt/dev_intelmq` as local repository copy. You can also use other directories as long as they are readable by other unprivileged users (e.g. home directories on Fedora can't be read by other users by default).
-`/opt/intelmq` is used as root location for IntelMQ installations, this is IntelMQ's default for this installation method. This directory is used for configurations (`/opt/intelmq/etc`), local states (`/opt/intelmq/var/lib`) and logs (`/opt/intelmq/var/log`).
+
+The traditional way to work with `intelmq` is to install it globally and have a separated user for running it. If you wish to separate your machine Python's libraries, you could alternatively use a virtual environment
+and the local user to run the `intelmq`. Please use the preferred way from instructions below.
```
Comment (sebix) [inline docs/dev/guide.rst]:
```suggestion
The traditional way to work with IntelMQ is to install it globally and have a separated user for running it. If you wish to separate your machine Python's libraries, e.g. for development purposes, you could alternatively use a Python virtual environment
and your local user to run IntelMQ. Please use your preferred way from instructions below.
```

File: docs/dev/guide.rst [hunk 4]
```diff
@@ -98,16 +146,20 @@ In case you developed a new bot, you need to update your current development ins
...
 
    cd /opt/dev_intelmq
-   ## necessary for pip metadata update and new executables:
    pip3 install -e .
 
+3. If you're using the global installation, an additional step of changing permissions and ownership is necessary:
+
+.. code-block:: bash
+
    find /opt/intelmq/ -type d -exec chmod 0770 {} \+
```
Comment (sebix) [inline docs/dev/guide.rst]: I think these steps (chown/chmod) are also done by `intelmqsetup`

File: docs/dev/guide.rst [hunk 5]
```diff
@@ -118,21 +170,22 @@ Now you can test run your new bot following this procedure:
...
 Testing
 =======
 
-Additional optional requirements
---------------------------------
+Additional test requirements
+----------------------------
 
-For the documentation tests one additional librariy is required: Cerberus. You can install it with pip:
+Libraries required for tests are listed in the `setup.py` file. You can install them with the pip:
```
Comment (sebix) [inline docs/dev/guide.rst]: either `with pip` or `with the pip program`

**Review by sebix — COMMENTED** [review]
Well done, thank you!

File: docs/dev/guide.rst (hunk 4 above)
Comment (kamil-certat) [reply docs/dev/guide.rst]: I'm unsure if it takes care all of the files and directories as here :/

I'd recommend leaving it for now.

File: docs/dev/guide.rst (hunk 1 above)
Comment (kamil-certat) [reply docs/dev/guide.rst]: Yes and no :D The goal is not to call it with sudo nor mix with the ownership. Anyway, I think it wasn't the best decision to leave the /opt/intelmq as a path. What do you think with the current change? It brings unfortunately a bit of complexity, but make it path-agnostic. I've tested installing IntelMQ totally as local user, without any ownership changes.

**Review by sebix — APPROVED** [review]

**Pull request merged**